### PR TITLE
SONAR-9328 optimized delete settings and manual measures when deleting a project

### PR DESCRIPTION
**Pull request by sns-seb**
based on #2127 

includes SONAR-9351

### CODE REVIEW

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/RecreateIndexManualMeasuresComponentUuid.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    context.execute(new CreateIndexBuilder(getDialect())
+      .setTable("manual_measures")
+      .setName("manual_measures_component_uuid")
+      .setUnique(false)
+      .addColumn(newVarcharColumnDefBuilder()
+        .setColumnName("component_uuid")
+        .setLimit(VarcharColumnDef.UUID_VARCHAR_SIZE)
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/RecreateIndexManualMeasuresComponentUuid.java]: MINOR Remove this use of "UUID_VARCHAR_SIZE"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/MakeManualMeasuresComponentUuidNotNullable.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  public MakeManualMeasuresComponentUuidNotNullable(Database db) {
+    super(db);
+  }
+
+  @Override
+  public void execute(DdlChange.Context context) throws SQLException {
+    context.execute(new AlterColumnsBuilder(getDialect(), "manual_measures")
+      .updateColumn(newVarcharColumnDefBuilder()
+        .setColumnName("component_uuid")
+        .setLimit(VarcharColumnDef.UUID_VARCHAR_SIZE)
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/MakeManualMeasuresComponentUuidNotNullable.java]: MINOR Remove this use of "UUID_VARCHAR_SIZE"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInManualMeasures.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      " where" +
+      "   mm.component_uuid is null" +
+      "   or not exists (" +
+      "     select" +
+      "       1" +
+      "     from projects p" +
+      "     where" +
+      "       p.uuid = mm.component_uuid" +
+      "       and p.scope = ?" +
+      "       and p.qualifier in (?,?,?,?)" +
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInManualMeasures.java]: `and p.enabled=true` should be added.

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInProperties.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      " where" +
+      "   s.resource_id is not null" +
+      "   and not exists (" +
+      "     select" +
+      "       1" +
+      "     from projects p" +
+      "     where" +
+      "       p.id = s.resource_id" +
+      "       and p.scope = ?" +
+      "       and p.qualifier in (?,?,?,?)" +
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInProperties.java]: `and p.enabled=true` should be added.

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/DbVersion65.java [hunk 5]
```diff
@@ -35,6 +35,7 @@ public void addSteps(MigrationStepRegistry registry) {
       .add(1705, "Make EVENTS.COMPONENT_UUID not nullable", MakeEventsComponentUuidNotNullable.class)
       .add(1706, "Recreate index EVENTS_COMPONENT_UUID", RecreateIndexEventsComponentUuid.class)
       .add(1707, "Ensure ISSUE.PROJECT_UUID is consistent", EnsureIssueProjectUuidConsistencyOnIssues.class)
-      .add(1708, "Clean orphans from PROJECT_LINKS", CleanOrphanRowsInProjectLinks.class);
+      .add(1708, "Clean orphans from PROJECT_LINKS", CleanOrphanRowsInProjectLinks.class)
+      .add(1709, "Clean orphans from SETTINGS", CleanOrphanRowsInProperties.class);
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/DbVersion65.java]: "Clean orphans from PROPERTIES" to be exact ?

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInManualMeasures.java (hunk 3 above)
Comment (sns-seb) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInManualMeasures.java]: done

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInProperties.java (hunk 4 above)
Comment (sns-seb) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v65/CleanOrphanRowsInProperties.java]: done

**Pull request merged**